pyodide-build/pyodide_build/install_xbuildenv.py

```diff
@@ -0,0 +1,44 @@
+import argparse
+import shutil
+import subprocess
+from pathlib import Path
+from sys import version_info
+
+from .common import get_make_flag, get_pyodide_root
+
+
+def make_parser(parser: argparse.ArgumentParser):
+    parser.description = (
+        "Install xbuild env.\n\n"
+        "Note: this is a private endpoint that should not be used "
+        "outside of the Pyodide Makefile."
+    )
+    return parser
+
+
+def main(args):
+    pyodide_root = get_pyodide_root()
+    host_site_packages = Path(get_make_flag("HOSTSITEPACKAGES"))
+    xbuildenv_path = pyodide_root / "xbuildenv"
+    major_minor = f"{version_info.major}.{version_info.minor}"
+    major_minor_patch = f"{major_minor}.{version_info.micro}"
+    include_path = (
+        pyodide_root
+        / f"cpython/installs/python-{major_minor_patch}/include/python{major_minor}/"
+    )
+    include_path.mkdir(exist_ok=True, parents=True)
+    shutil.copytree(xbuildenv_path / "python-include", include_path, dirs_exist_ok=True)
+    host_site_packages.mkdir(exist_ok=True, parents=True)
+    subprocess.run(
+        [
+            "pip",
+            "install",
+            "-t",
+            host_site_packages,
+            "-r",
+            xbuildenv_path / "requirements.txt",
+        ]
+    )
+    shutil.copytree(
+        xbuildenv_path / "site_packages", host_site_packages, dirs_exist_ok=True
+    )
```

Comment:
Hmm, but so NumPy would be in the `requirements.txt`, for instance, right? I'm not sure if installing the unpatched numpy version and then overwriting site_packages with the host_site_packages is that reliable and there isn't a chance we would break something.

Reply:
It's a bit sketchy, but I think it's correct and reproducible. We've locked the two numpy versions to be the same. When we update numpy, we'll update the lock too.

This logic is essentially unchanged here: since #2256, when we build numpy we've been downloading a copy of numpy from pip and writing over `_npyconfig.h`. Before that, we built a fresh native copy of numpy and then wrote over `_npyconfig.h`. The difference here is that we are planning to download both `numpy` and `_npyconfig.h` and paste them together.